AmplifyPlugins/DataStore/Sources/AWSDataStorePlugin/Sync/MutationSync/AWSMutationDatabaseAdapter/AWSMutationDatabaseAdapter+MutationEventIngester.swift

```diff
@@ -33,21 +33,6 @@ extension AWSMutationDatabaseAdapter: MutationEventIngester {
     func resolveConflictsThenSave(mutationEvent: MutationEvent,
                                   storageAdapter: StorageEngineAdapter,
                                   completion: @escaping (Result<MutationEvent, DataStoreError>)->Void) {
-
-        // We don't want to query MutationSync<AnyModel> because a) we already have the model, and b) delete mutations
-        // are submitted *after* the delete has already been applied to the local data store, meaning there is no model
-        // to query.
-        var mutationEvent = mutationEvent
-        do {
-            // TODO: Refactor this so that it's clear that the storage engine is not responsible for setting the version
-            // perhaps as simple as renaming to `submit(unversionedMutationEvent:)` or similar
-            let syncMetadata = try storageAdapter.queryMutationSyncMetadata(for: mutationEvent.modelId,
-                                                                               modelName: mutationEvent.modelName)
-            mutationEvent.version = syncMetadata?.version
```

Comment:
Down below on line 196 creates a new MutationEvent based on a "candidate" mutation event and sets the version from the candidate. This line isn't needed anymore, right? If not needed, we should remove it as well since it will be confusing to the reader in the future to understand why we have logic around the version field in this file at all.

Reply:
do you mean the logic inside `getResolvedEvent`? This is about merging 2 MutationEvents into one. Even the version is not used when submitting to AppSync. As we still keep the merging logic, it makes sense to take the candidate version over the old mutation event version. It might help with debugging.

Reply:
Yes inside `getResolvedEvent`, is `resolvedEvent.version = candidate.version` always setting nil? If we remove the current code block (from lines 36-50), would the candidate version always be nil? If so, it is confusing why we will set the version during the merging logic

Reply:
oh, I see. The candidate was created in this block of code. Yeah, I will remove that version assignment.